src/network/network_chat_gui.cpp

```diff
@@ -168,28 +173,21 @@ void NetworkUndrawChatMessage()
 		/* And make sure it is updated next time */
 		VideoDriver::GetInstance()->MakeDirty(x, y, width, height);
 
+		_chatmessage_dirty_time = std::chrono::steady_clock::now();
 		_chatmessage_dirty = true;
 	}
 }
 
 /** Check if a message is expired. */
 void NetworkChatMessageLoop()
 {
-	for (uint i = 0; i < MAX_CHAT_MESSAGES; i++) {
-		ChatMessage *cmsg = &_chatmsg_list[i];
-		if (cmsg->message[0] == '\0') continue;
-
+	auto now = std::chrono::steady_clock::now();
+	for (auto it = _chatmsg_list.begin(); it != _chatmsg_list.end(); it++) {
```

Comment:
`for (auto it : _chatmsg_list)` ?

Learning C++, I can be dead-wrong.

Reply:
Lost in a rebase conflict, oops.